batch/test/test_batch.py

```diff
@@ -1070,3 +1070,101 @@ def test_job_private_instance_cancel(client: BatchClient):
     b.cancel()
     status = j.wait()
     assert status['state'] == 'Cancelled', str((status, b.debug_info()))
+
+
+def test_update_batch_no_deps(client: BatchClient):
+    bb = client.create_batch()
+    bb.create_job(DOCKER_ROOT_IMAGE, ['false'])
+    bb.submit()
+    j = bb.create_job(DOCKER_ROOT_IMAGE, ['true'])
+    b = bb.submit()
+    status = j.wait()
+
+    assert status['state'] == 'Success', str((status, b.debug_info()))
+
+
+def test_empty_update(client: BatchClient):
+    bb = client.create_batch()
+    bb.create_job(DOCKER_ROOT_IMAGE, ['true'])
+    b = bb.submit()
+    b = bb.submit()
+    status = b.wait()
+
+    assert status['state'] == 'success', str((status, b.debug_info()))
```

Comment:
Let's right the following test case:

1. Submit with an hour long sleep job, `j1`
2. Submit another update with an always-run `true` job, `j2` which is a child of `j1`
3. Verify that `j1` is `Running` and `j2` is `Pending`
4. Cancel `j1`
5. Wait on `j2` and verify that it completed successfully

Reply:
Another similar test case that just does a sanity check that jobs from different updates aren't somehow serialized

1. Submit an hour long sleep job, `j1`
2. Submit another update with a `true` job `j2`
3. Wait on `j2` and see that it completes successfully
4. Verify that `j1` is still `Running`
5. Cancel `j1` so the job ends and developers aren't sad waiting for testing to finish